code/modules/mob/living/carbon/xenomorph/castes/drone/drone.dm

```diff
@@ -15,3 +15,9 @@
 	inherent_verbs = list(
 		/mob/living/carbon/xenomorph/proc/vent_crawl,
 	)
+
+/mob/living/carbon/xenomorph/drone/on_death()
+	var/datum/action/xeno_action/activable/essence_link/essence_link_action = actions_by_path[/datum/action/xeno_action/activable/essence_link]
+	if(essence_link_action)
+		essence_link_action.end_ability()
+	return ..()
```

Comment:
this should be on the ability using signals and not on_death

Reply:
Turns out this is infact redundant, that is already the case